### ISSUE TITLE
Support networks that use `type: "dhcp"`

### ISSUE THREAD

**Issue by TronPaul** [oscar-stack/vagrant-hosts#45]
## Setup:

```
Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  config.vm.define :thing1 do |box|
    box.vm.box = "ubuntu/trusty64"
    box.vm.hostname = "thing1"
    box.vm.network :private_network, type: "dhcp"
    box.vm.provision :hosts
  end
end
```
## What happens:

The hosts file added to the guests looks like

```
 thing1
```
## What should happen:

The host file looks like

```
<DHCP-ADDRESS> thing1
```

where <DHCP-ADDRESS> is the IP address assigned.

---

I went digging for if the IP of the guest vm could be grabbed from somewhere in vagrant, but I wasn't able to find where that might be. Until someone smarter than me figures out how to do that, I think the best solution is to ignore guest networks that are using DHCP to get their IP addresses for now.

**Comment by glenux** [comment]
Some clues here : https://github.com/smdahlen/vagrant-hostmanager/issues/86

Seems to work from ruby with `VBoxManage guestproperty get #{vm.id} "/VirtualBox/GuestInfo/Net/1/V4/IP"`.split()[1]

**Comment by brijs** [comment]
DHCP isn't supported. I believe vagrant hosts plugin only supports `private_network`, and only when `ip` address is explicitly specified in the Vagrantfile. See [vagrant-hosts/address.rb](https://github.com/oscar-stack/vagrant-hosts/blob/95eabdc59e7898767291093776db2073086ac2af/lib/vagrant-hosts/addresses.rb#L47-L48)

**Comment by Sharpie** [comment]
> DHCP isn't supported. I believe vagrant hosts plugin only supports private_network

This is correct. `vagrant-hosts` only pulls from `private_network` configuration. Added a note to the README that clarifies this and marking this issue as a feature request.
